Mesh_3/include/CGAL/Mesh_complex_3_in_triangulation_3.h

```diff
@@ -547,7 +547,105 @@ class Mesh_complex_3_in_triangulation_3 :
                                  Vertex_iterator_not_in_complex(*this));
   }
 
+  std::ostream& write_edges_in_complex(std::ostream& os,
+                                      const Unique_hash_map<Vertex_handle, std::size_t >& vertex_index_map) const
+  {
+    if(is_ascii(os))
+    {
+      os << number_of_edges_in_complex()<<std::endl;
+    }
+    else
+    {
+      write(os, number_of_edges_in_complex());
+    }
+    if(is_ascii(os))
+    {
+      for(auto it = edges_.begin(); it != edges_.end(); ++it)
+      {
+        os<<vertex_index_map[it->left]<<" "<<vertex_index_map[it->right]<< " "<< CGAL::oformat(it->info)<<std::endl;
+      }
+    }
+    else
+    {
+      if(is_ascii(os))
+      {
+        for(auto it = edges_.begin(); it != edges_.end(); ++it)
+        {
+          write(os, vertex_index_map[it->left]);
+          write(os, vertex_index_map[it->right]);
+          os << CGAL::oformat(it->info);
+        }
+      }
```

Comment:
The `if(is_ascii(os))` line 570 is in the `else` part of another `if(is_ascii(os))`.

And I suggest you regroup everything in one `if(is_ascii(os))/else`.